Resources/config/services.xml

```diff
@@ -5,26 +5,44 @@
            xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">
 
     <services>
+        <service id="Shivas\VersioningBundle\Formatter\FormatterInterface"
+                 alias="Shivas\VersioningBundle\Formatter\GitDescribeFormatter"/>
+        <service id="Shivas\VersioningBundle\Writer\WriterInterface"
+                 alias="Shivas\VersioningBundle\Writer\VersionWriter"/>
+
         <service id="Shivas\VersioningBundle\Service\VersionManager">
+            <argument type="service" id="Symfony\Component\Cache\Adapter\AdapterInterface"/>
```

Comment:
you should not get the main cache pool (which is the `cache.app`, not cleaned on deployment btw), but create your own cache pool (so that keys in it are isolated from the cache keys of the project to avoid conflicts). And this cache pool should be based on `cache.system` if you want it to be cleared automatically on deployment.
This is very easy: https://github.com/symfony/symfony/blob/v4.0.7/src/Symfony/Bundle/FrameworkBundle/Resources/config/cache.xml#L18-L20